### DISCUSSION TITLE
FUXA as modbus TCP slave

### DISCUSSION THREAD

**Discussion by vasiukoff** [frangoteam/FUXA#987] [Q&A] [answered]
Hi!

Is it possible to use FUXA as modbus TCP slave? I have an equipment which sends the data via modbus TCP. My idea is to point this equipment to FUXA and collect the data. 
If FUXA does not support this feature itself, may be some workarounds is possible? 
Thanks :)

**Comment by vasiukoff** [comment, marked as answer]
Since no one answered, I did everything myself. :)
I found a fairly old Modbus TCP server project written in Python2. I added SlaveID filtering and made a number of cosmetic additions that allow this server to work if a stupid gateway from RS485 to TCP is used. 
I also converted it to Python3. Tested on Linux, but it should work on Windows too.

The solution now looks like this:

FUXA ----> Slave server <---- Equipment

I'm currently testing this solution. 
Everyone who is interested - wellcome. :)

**Reply by concept10** [reply]
The initial post describes equipment that uses Modbus TCP. Most all devices you shall encounter in the industry are to be used as Modbus slaves. In this case, FUXA would be the Modbus TCP master polling data from the equipment using the node address and proper addressing of variables that are desired to be read. I'm not sure what's being missed here.

**Comment by jingshui127** [comment]
MQTT
